packages/dds/merge-tree/src/collections/list.ts

```diff
@@ -3,142 +3,127 @@
  * Licensed under the MIT License.
  */
 
-export function ListRemoveEntry<U>(entry: List<U>): List<U> | undefined {
-    if (entry === undefined) {
-        return undefined;
-    } else if (entry.isHead) {
-        return undefined;
-    } else {
-        entry.next.prev = entry.prev;
-        entry.prev.next = entry.next;
-        entry.next = deadhead;
-        entry.prev = deadhead;
-    }
-    return (entry);
-}
+import { UsageError } from "@fluidframework/container-utils";
 
-function ListMakeEntry<U>(data: U): List<U> {
-    return new List<U>(false, data);
+export interface ListNode<T> {
+    readonly list: List<T> | undefined;
+    readonly data: T;
+    readonly next: ListNode<T> | undefined;
+    readonly prev: ListNode<T> | undefined;
 }
 
-export function ListMakeHead<U>(): List<U> {
-    return new List<U>(true, undefined);
+export interface ListNodeRange<T> {
+    first: ListNode<T>;
+    last: ListNode<T>;
 }
 
-export class List<T> {
-    public next: List<T>;
-    public prev: List<T>;
-
-    constructor(public isHead: boolean, public data: T | undefined) {
-        this.prev = this;
-        this.next = this;
-    }
-
-    public clear(): void {
-        if (this.isHead) {
-            this.prev = this;
-            this.next = this;
+class HeadNode<T> {
+    public _next: HeadNode<T> | DataNode<T> = this;
+    public _prev: HeadNode<T> | DataNode<T> = this;
+    public headNode: HeadNode<T>;
+    private readonly _list?: List<T>;
+    constructor(list: List<T> | undefined) {
+        this.headNode = this;
```

Comment:
is this different from the sugar you use to initialize `_next` and `_prev`? assuming not, i'd prefer consistent style

Reply:
yes. as when called by the DataNode which inherit from head node they will pass this and set it as whatever head they are bound to.

Reply:
ha. but looks like i don't even use it that way 🤦‍♂️. i wrote this a while back, and just peeling off changes. i'll make it more consistent